src/main/java/org/spongepowered/api/world/explosion/ResistanceCalculator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world.explosion;
+
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.block.BlockState;
+
+/**
+ * Calculates a Block's resistance to a {@link Explosion}.
+ */
+@FunctionalInterface
+public interface ResistanceCalculator {
+
+    /**
+     * Sets the explosion resistance for a block at a given location
+     * during an {@link Explosion}.
```

Comment:
What am I supposed to understand about setting the resistance for a block at a given location?

Is it that I decrement the returned value from the original resistance?

Can I chain on the vanilla calculator?

Is it possible to return some bounds?

What are the numbers meaning? Do I return `0` to say "don't break me" or do I use `1f` to say "I'm impenetrable? Or is it a resistance factor based on `0-10`?